driver/src/contracts/mod.rs

```diff
@@ -26,3 +28,36 @@ fn method_defaults(key: PrivateKey, network_id: u64) -> Result<MethodDefaults> {
 
     Ok(defaults)
 }
+
+const EXPECTED_ERRORS: &[&str; 3] = &[
+    "New objective doesn\'t sufficiently improve current solution",
+    "Claimed objective doesn't sufficiently improve current solution",
+    "SafeMath: subtraction overflow",
+];
+
+pub fn extract_benign_submission_failure(err: &Error) -> Option<String> {
```

Comment:
Nit: This can probably return `Option<'static str>` or an `enum ExpectedError` (which implements a conversion to the exact error string). But I see that this would make the code a bit longer and this isn't a performance concern so feel free to disregard this comment.

Reply:
Thanks for the suggestion. It actually pushed me into making this a more typesafe top-level return type of the solution submission.